Document the ability to deprecate by date (#69833)
ansible/ansible: docs/docsite/rst/dev_guide/module_lifecycle.rst

4. Add ``deprecated:`` to the documentation with the following sub-values:

>   :removed_in: A ``string``, such as ``"2.10"``; the version of Ansible where the module will be replaced with a docs-only module stub. Usually current release +4.

Reviewer: ```suggestion
  :removed_in: A ``string``, such as ``"2.10"``; the version of Ansible where the module will be replaced with a docs-only module stub. Usually current release +4. Mutually exclusive with :removed_by_date:.
```

>   :remove_by_date: (Added in Ansible 2.10). An ISO 8601 formatted date when the module will be removed. Usually 2 years from the date the module is deprecated. This is mutually exclusive to :remove_by_date:.

Reviewer: ```suggestion
  :remove_by_date: (Added in Ansible 2.10). An ISO 8601 formatted date when the module will be removed. Usually 2 years from the date the module is deprecated. Mutually exclusive with :removed_in:.
```
